tests/rest/test_rest.py

```diff
@@ -37,11 +37,16 @@
 
 
 def get_data_from_url(url, result, index):
-    response = urllib.request.urlopen(urllib.request.Request(url))
-    body = response.read()
-    data = json.loads(body)
-    result[index] = data
 
+    try:    
+        response = urllib.request.urlopen(urllib.request.Request(url))
+        body = response.read()
+        data = json.loads(body)
+        result[index] = data
+
+    
+    except urllib.error.HTTPError as e:
+        print(e.code)   
```

Comment:
why is raise exception not good?

Reply:
+1, why catching the exception?

Reply:
I expect to just record the error here then raise one exception in main thread(on top of all subprocess), so for each subprocess, its job is either to get data(success) or to print error log(fail).

Reply:
> then raise one exception in main thread

@tttttangTH  I didn't follow how this is implemented. 

Let me ask a question: Will the CI fail if a `urllib.error.HTTPError` exception is raised in this method?

Reply:
Yes, CI will fail.

Reply:
> Yes, CI will fail.

@tttttangTH Could you explain how it works?
I don't see how it can make the CI fail, you are just printing a message...

Reply:
There is an array `result`  which is used to record the response of each thread. if one thread gets HTTP error, the value in `result` will still be `None`, we will check this in main thread.